src/device/mod.rs

```diff
@@ -435,10 +426,6 @@ impl Device {
 
         telio_log_info!("libtelio is starting up with features : {:?}", features);
 
-        #[cfg(any(target_os = "macos", target_os = "ios", target_os = "tvos"))]
-        NETWORK_PATH_MONITOR_START
-            .call_once(telio_sockets::protector::platform::setup_network_path_monitor);
```

Comment:
Removing this and moving it into the `NativeProtector::new` might significantly reduce the time the OS has to generate the notification. I think this call should remain here in some form.

Reply:
We can create the protector in the `Device::new()`. Do you think that makes sense?

Reply:
I've moved the socket pool creation to `Device::new()`